Doesn't build with armv7 on openSUSE Tumbleweed

### What is the bug or the crash?

As you can see here https://build.opensuse.org/package/live_build_log/Application:Geo:Staging/qgis/openSUSE_Factory_ARM/armv7l it does not build:

```
[  655s] In file included from /home/abuild/rpmbuild/BUILD/qgis-3.30.1/src/core/geometry/qgsabstractgeometry.h:27,
[  655s]                  from /home/abuild/rpmbuild/BUILD/qgis-3.30.1/src/core/geometry/qgspoint.h:23,
[  655s]                  from /home/abuild/rpmbuild/BUILD/qgis-3.30.1/src/core/geometry/qgsbox3d.h:26,
[  655s]                  from /home/abuild/rpmbuild/BUILD/qgis-3.30.1/src/core/metadata/qgslayermetadata.h:24,
[  655s]                  from /home/abuild/rpmbuild/BUILD/qgis-3.30.1/src/core/qgsmaplayer.h:38,
[  655s]                  from /home/abuild/rpmbuild/BUILD/qgis-3.30.1/src/core/mesh/qgsmeshlayer.h:25,
[  655s]                  from /home/abuild/rpmbuild/BUILD/qgis-3.30.1/src/core/mesh/qgsmeshcalcutils.h:35,
[  655s]                  from /home/abuild/rpmbuild/BUILD/qgis-3.30.1/src/core/mesh/qgsmeshcalcnode.h:33,
[  655s]                  from mesh/qgsmeshcalcparser.yy:19:
[  655s] /home/abuild/rpmbuild/BUILD/qgis-3.30.1/src/core/geometry/qgswkbptr.h:83:29: error: 'const QgsWkbPtr& QgsWkbPtr::operator>>(qsizetype&) const' cannot be overloaded with 'const QgsWkbPtr& QgsWkbPtr::operator>>(int&) const'
[  655s]    83 |     inline const QgsWkbPtr &operator>>( qsizetype &r ) const { int v; read( v ); r = v; return *this; } SIP_SKIP
[  655s]       |                             ^~~~~~~~
[  655s] /home/abuild/rpmbuild/BUILD/qgis-3.30.1/src/core/geometry/qgswkbptr.h:80:29: note: previous declaration 'const QgsWkbPtr& QgsWkbPtr::operator>>(int&) const'
[  655s]    80 |     inline const QgsWkbPtr &operator>>( int &v ) const { read( v ); return *this; } SIP_SKIP
[  655s]       |                             ^~~~~~~~
[  655s] /home/abuild/rpmbuild/BUILD/qgis-3.30.1/src/core/geometry/qgswkbptr.h:100:23: error: 'QgsWkbPtr& QgsWkbPtr::operator<<(qsizetype)' cannot be overloaded with 'QgsWkbPtr& QgsWkbPtr::operator<<(int)'
[  655s]   100 |     inline QgsWkbPtr &operator<<( qsizetype r ) { int v = r; write( v ); return *this; } SIP_SKIP
[  655s]       |                       ^~~~~~~~
[  655s] /home/abuild/rpmbuild/BUILD/qgis-3.30.1/src/core/geometry/qgswkbptr.h:97:23: note: previous declaration 'QgsWkbPtr& QgsWkbPtr::operator<<(int)'
[  655s]    97 |     inline QgsWkbPtr &operator<<( int v ) { write( v ); return *this; } SIP_SKIP
[  655s]       |                       ^~~~~~~~
```

I assume that qsizetype is int on that system, so that's not possible to have this function with arguments qsizetype und int. Only one of them may exist.

### Steps to reproduce the issue

Build for arm7 (I assume any system with qsizetype=int has same results).

Any additional details needed (gcc versions, lib versions, ...) can be seen in the linked build log.

### Versions

3.30.1

### Supported QGIS version

- [ ] I'm running a supported QGIS version according to the roadmap.

### New profile

- [ ] I tried with a new QGIS profile

### Additional context

_No response_

--- a/src/core/geometry/qgswkbptr.h
+++ b/src/core/geometry/qgswkbptr.h
@@ -77,12 +77,13 @@ class CORE_EXPORT QgsWkbPtr
 
     inline const QgsWkbPtr &operator>>( double &v ) const { read( v ); return *this; } SIP_SKIP
     inline const QgsWkbPtr &operator>>( float &r ) const { double v; read( v ); r = v; return *this; } SIP_SKIP
-    inline const QgsWkbPtr &operator>>( int &v ) const { read( v ); return *this; } SIP_SKIP
-#if (!defined(Q_OS_ANDROID) && !defined(Q_PROCESSOR_X86_32)) || (defined(Q_OS_ANDROID) && defined(__aarch64__))
-    //! Reads an integer value into a qsizetype
-    inline const QgsWkbPtr &operator>>( qsizetype &r ) const { int v; read( v ); r = v; return *this; } SIP_SKIP
-#endif
-    inline const QgsWkbPtr &operator>>( unsigned int &v ) const { read( v ); return *this; } SIP_SKIP
+    //! Reads an integer value into a qint32
+    inline const QgsWkbPtr &operator>>( qint32 &v ) const { read( v ); return *this; } SIP_SKIP
+    //! Reads an integer value into a longlong
+    inline const QgsWkbPtr &operator>>( qint64 &r ) const { quint32 v; read( v ); r = v; return *this; } SIP_SKIP
+    //! Reads an unsigned integer value
+    inline const QgsWkbPtr &operator>>( quint32 &v ) const { read( v ); return *this; } SIP_SKIP
+    //! Reads an char value
     inline const QgsWkbPtr &operator>>( char &v ) const { read( v ); return *this; } SIP_SKIP
     inline const QgsWkbPtr &operator>>( QgsWkbTypes::Type &v ) const { read( v ); return *this; } SIP_SKIP
 
@@ -91,13 +92,11 @@ class CORE_EXPORT QgsWkbPtr
     //! Writes a float to the pointer
     inline QgsWkbPtr &operator<<( float r ) { double v = r; write( v ); return *this; } SIP_SKIP
     //! Writes an int to the pointer
-    inline QgsWkbPtr &operator<<( int v ) { write( v ); return *this; } SIP_SKIP
-#if (!defined(Q_OS_ANDROID) && !defined(Q_PROCESSOR_X86_32)) || (defined(Q_OS_ANDROID) && defined(__aarch64__))
-    //! Writes a size as int to the pointer
-    inline QgsWkbPtr &operator<<( qsizetype r ) { int v = r; write( v ); return *this; } SIP_SKIP
-#endif
+    inline QgsWkbPtr &operator<<( qint32 v ) { write( v ); return *this; } SIP_SKIP
+    //! Writes a longlong as int to the pointer
+    inline QgsWkbPtr &operator<<( qint64 r ) { quint32 v = r; write( v ); return *this; } SIP_SKIP
     //! Writes an unsigned int to the pointer
-    inline QgsWkbPtr &operator<<( unsigned int v ) { write( v ); return *this; } SIP_SKIP
+    inline QgsWkbPtr &operator<<( quint32 v ) { write( v ); return *this; } SIP_SKIP
     //! Writes a char to the pointer
     inline QgsWkbPtr &operator<<( char v ) { write( v ); return *this; } SIP_SKIP
     //! Writes a WKB type value to the pointer

